### PR TITLE
Fix save project (backup pipeline fails on working with symlinks)

### PR DESCRIPTION
**Pull request by brusherru**
There is no issue. 🙈 

Fix and little refactoring of the backup process in the `xod-fs`.

Also detected another bug: when tries to save project many times, before it complete, it occurs bug. It fixed too by preventing start new save process before previous was finished.

### CODE REVIEW

**Comment by nkrkv** [comment]
Does not fixed the problem for me

**Comment by brusherru** [comment]
@nkrkv check it out, please!

File: packages/xod-fs/src/backup.js [hunk 1]
```diff
@@ -23,17 +53,18 @@ export class Backup {
   }
 
   make() {
-    return new Promise((resolve, reject) => {
-      if (!this.isDataExist) { resolve('data is not exist'); return; }
-      if (!this.isTempExist) { fs.mkdirSync(this.path.temp); }
-      if (!this.isDataTempExist) { fs.mkdirSync(this.path.dataTemp); }
+    if (!this.isDataExist) {
+      return Promise.reject(
+        new Error('Can not make a backup: Data is not exist')
```
Comment (nkrkv) [inline packages/xod-fs/src/backup.js]: Suggest:

```js
`Backup failed: '${this.path.data}' does not exist`
```

**Comment by nkrkv** [comment]
The `this.setState({ saveInProgress: false })` is very fragile. We need a more bullet-proof solution to the problem.

**Pull request closed without merging**